src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-storage-class-details/rgw-storage-class-details.component.ts

```diff
@@ -48,38 +50,51 @@ export class RgwStorageClassDetailsComponent implements OnChanges, OnInit {
   restoreStorageClassText = RESTORE_STORAGE_CLASS_TEXT;
   zoneGroupText = ZONEGROUP_TEXT;
   groupedACLs: GroupedACLs = {};
+  localStorageClassDetails = { zone_name: '', data_pool: '' };
+  loading = false;
+
+  private rgwZoneService = inject(RgwZoneService);
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes['selection']) {
-      this.storageDetails = {
-        zonegroup_name: this.selection?.zonegroup_name,
-        placement_targets: this.selection?.placement_targets,
-        access_key: this.selection?.access_key,
-        secret: this.selection?.secret,
-        target_path: this.selection?.target_path,
-        tier_type: this.selection?.tier_type,
-        multipart_min_part_size: this.selection?.multipart_min_part_size,
-        multipart_sync_threshold: this.selection?.multipart_sync_threshold,
-        host_style: this.selection?.host_style,
-        retain_head_object: this.selection?.retain_head_object,
-        allow_read_through: this.selection?.allow_read_through,
-        glacier_restore_days: this.selection?.glacier_restore_days,
-        glacier_restore_tier_type: this.selection?.glacier_restore_tier_type,
-        restore_storage_class: this.selection?.restore_storage_class,
-        read_through_restore_days: this.selection?.read_through_restore_days
-      };
+    if (
+      changes['selection'] &&
+      changes['selection'].currentValue?.tier_type?.toLowerCase() === TIER_TYPE.LOCAL &&
+      changes['selection'].firstChange
+    ) {
+      // The idea here is to not call the API if we already have the zone_name and data_pool
+      // When the details view is expanded and table refreshes data then this API should not be called again
+      const { zone_name, data_pool } = this.localStorageClassDetails;
+      if (!zone_name || !data_pool) {
+        this.getZoneInfo();
+      }
     }
   }
 
   ngOnInit() {
-    this.groupedACLs = this.groupByType(this.selection.acl_mappings);
+    this.groupedACLs = this.groupByType(this.selection?.acl_mappings);
   }
 
   isTierMatch(...types: string[]): boolean {
     const tier_type = this.selection.tier_type?.toLowerCase();
     return types.some((type) => type.toLowerCase() === tier_type);
   }
 
+  getZoneInfo() {
```

Comment:
I don't think it's necessary to fetch all zones again, it's already being done on the listing.

https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-storage-class-list/rgw-storage-class-list.component.ts#L118